File: testClients/pathCheck/AI.cpp
#include "AI.h"
#include "util.h"

#include <iostream>
using namespace std;
AI::AI(Connection* conn) : BaseAI(conn) {}

const char* AI::username()
{
  return "Shell AI";
}

const char* AI::password()
{
  return "password";
}

//This function is run once, before your first turn.
void AI::init()
{
  srand(time(NULL));
  cout<<"Initialization called"<<endl;
  land = vector<vector<bool> > (mapSize(),vector<bool>(mapSize(),false));
  port = vector<vector<int> > (mapSize(),vector<int>(mapSize(),-1));
  map = vector<vector<bool> > (mapSize(),vector<bool>(mapSize(),false));


  for(size_t i=0;i<tiles.size();i++)
  {
    land[tiles[i].x()][tiles[i].y()]= tiles[i].type()==0;
  }
  for(size_t i=0;i<ports.size();i++)
  {
    port[ports[i].x()][ports[i].y()]=ports[i].owner();
  }
  vector<Tile*> p = getPath(ports[0].x(), ports[0].y(), ports[1].x(), ports[1].y(), 1);
  for(int i = 0; i < p.size(); i++)
  {
    int x = p[i]->x();
    int y = p[i]->y();
    map[x][y] = true;
  }
  for(size_t i=0;i<land.size();i++)
  {
    for(size_t j=0;j<land[i].size();j++)
    {
      if(port[i][j] != -1)
        cout << port[i][j] << " ";
      else if(map[i][j])
        cout << "- ";
      else
        cout<<(land[i][j]?"X":".")<<" ";
    }
    cout<<endl;
  }
}

//This function is called each time it is your turn.
//Return true to end your turn, return false to ask the server for updated information.
bool AI::run()
{
  cout<<"Starting turn: "<<turnNumber()<<endl;
  
  return true;
}

//This function is run once, after your last turn.
void AI::end()
{
  cout<<"End Called"<<endl;
}

File: testClients/javaCheck/AI.java
import com.sun.jna.Pointer;
import java.util.*;
///The class implementing gameplay logic.
class AI extends BaseAI
{
  public String username()
  {
    return "Shell AI";
  }
  public String password()
  {
    return "password";
  }
  Random generator = new Random();
  int xoff[] = {0, 1, 0, -1};
  int yoff[] = {1, 0, -1, 0};

  //This function is called each time it is your turn
  //Return true to end your turn, return false to ask the server for updated information
  public boolean run()
  {
    System.out.println("Turn Number: "+turnNumber());
    printPirates();
    int myPort=0;
    while(ports[myPort].getOwner() != playerID())myPort++;
    for(int i=0;i<players[playerID()].getGold();i+=pirateCost())
    {
    //  ports[myPort].createPirate();
    }
    
    ArrayList<Integer> enemies= new ArrayList<Integer>();
    for(int i=0;i<pirates.length;i++)
    {
      if(pirates[i].getOwner() != playerID())
      {
        enemies.add(i);
      }
    }
    
    for(int i=0;i<pirates.length;i++)
    {
      if(pirates[i].getOwner() == playerID())
      {
        int target = generator.nextInt(enemies.size());
        target=0;
        if(pirates[enemies.get(target)].getOwner()==playerID())
        {
          System.out.println("FUCK");
        }
        Tile path[] = getPath(pirates[i].getX(), pirates[i].getY(), 
                              pirates[enemies.get(target)].getX(), pirates[enemies.get(target)].getY(),0);
        if(path.length>1)
        {
          System.out.println("Move");
          pirates[i].move(path[0].getX(),path[0].getY());
        }
        else if(path.length==1)
        {
          System.out.println("Attack");
          pirates[i].attack(pirates[enemies.get(target)]);
        }
      }
    }
    return true;
  }
  private void printPirates()
  {
    for(int i=0;i<pirates.length;i++)
    {
      if(playerID()==pirates[i].getOwner())
      {
        printPirate(i);
        System.out.println();
      }
      
    }
  }
  
  private void printPirate(int i)
  {
    System.out.print("pirates["+i+"]=(id "+pirates[i].getId()
                                 +") (x "+pirates[i].getX()
                                 +") ("+pirates[i].getY()
                                 +") ("+pirates[i].getOwner()
                                 +") ("+pirates[i].getHealth()
                                 +") ("+pirates[i].getStrength()
                                 +")"); 
  }

  //This function is called once, before your first turn
  public void init() { }

  //This function is called once, after your last turn
  public void end() {}
  
  
  public AI(Pointer c)
  {
    super(c);
  }
}
